examples/routing_service/routing_service_file_adapter/cpp/FileStreamReader.cxx

```diff
@@ -0,0 +1,144 @@
+/*****************************************************************************/
+/*         (c) Copyright, Real-Time Innovations, All rights reserved.        */
+/*                                                                           */
+/*         Permission to modify and use for internal purposes granted.       */
+/* This software is provided "as is", without warranty, express or implied.  */
+/*                                                                           */
+/*****************************************************************************/
+
+#include <sstream>
+#include <thread>
+
+#include "FileStreamReader.hpp"
+
+using namespace rti::community::examples;
+using namespace dds::core::xtypes;
+
+const std::string FileStreamReader::INPUT_FILE_PROPERTY_NAME = "example.adapter.input_file";
+const std::string FileStreamReader::SAMPLE_PERIOD_PROPERTY_NAME = "example.adapter.sample_period_sec";
+
+void FileStreamReader::file_reading_thread()
+{
+    while (!stop_thread_) {
+        if (input_file_stream_.is_open()) {
+            std::getline(input_file_stream_, buffer_);
+
+            /** 
+             * Here we notify routing service, that there is data available 
+             * on the StreamReader, triggering a call to take().
+             */
+            if (!input_file_stream_.eof()) {
+                reader_listener_->on_data_available(this);
+            } else {
+                stop_thread_ = true;
+            }
+        }
+
+        std::this_thread::sleep_for(std::chrono::seconds(sampling_period_));
+    }
+    std::cout << "Reached end of stream for file " << input_file_name_ << std::endl;
+    file_connection_->dispose_discovery_stream(stream_info_);
+}
+
+FileStreamReader::FileStreamReader(
+        FileConnection *connection, 
+        const StreamInfo &info,
+        const PropertySet &property,
+        StreamReaderListener *listener)
+        : filereader_thread_(),
+        sampling_period_(1), 
+        stop_thread_(false)
+{
+    stream_info_ = new StreamInfo(
```

Comment:
this one we don't need to make a heap copy; make it a stack member

Reply:
I can't since its used in the file_reading_thread()

Reply:
why is that a problem? that shouldn't make any difference.

Reply:
Changed it to be a class member instead of a pointer.